dcap/types/src/certification_data.rs

```diff
@@ -273,10 +352,15 @@ mod test {
         ";
 
     #[test]
-    fn iterate_over_a_empty_certification_data() {
-        let cert_data = [0u8; MIN_CERT_DATA_SIZE];
+    fn iterate_over_a_empty_pck_cert_chain() {
+        let mut cert_data = [0u8; MIN_CERT_DATA_SIZE];
+        cert_data[0] = 5;
         let certification_data = CertificationData::try_from(cert_data.as_slice()).unwrap();
-        let cert_iter = certification_data.into_iter();
+        let cert_chain = match certification_data {
+            CertificationData::PckCertChain(cert_chain) => cert_chain,
+            _ => panic!("expected a PckCertChain"),
+        };
```

Comment:
This feels kind of wordy to me to get the expected variant, but `if let` seemed even more wordy.
Not sure if there is a more concise way to get at the underlying variant

Reply:
I've come across this, and I think that this is the best way to do it.

Reply:
I slightly better way is coming... https://blog.rust-lang.org/2022/11/03/Rust-1.65.0.html#let-else-statements

Reply:
This repo is built against `stable`, `beta` (stable-next, without `feature()`), and a particular nightly, so a slightly better way is technically already here :-)

Reply:
`nightly-2022-07-22` 
```shell
error[E0658]: `let...else` statements are unstable
   --> dcap/types/src/certification_data.rs:356:9
    |
356 | /         let CertificationData::PckCertificateChain(cert_chain) = certification_data else {
357 | |             panic!("expected a PckCertChain");
358 | |         };
    | |__________^
    |
    = note: see issue #87335 <https://github.com/rust-lang/rust/issues/87335> for more information
    = help: add `#![feature(let_else)]` to the crate attributes to enable
```
:*(

Reply:
Just update that nightly, we're only using it here (mobilecoin.git is still on 2022-04-28 or similar).